numpy/f2py/cfuncs.py

```diff
@@ -647,8 +647,8 @@
             n = PyArray_NBYTES(arr);
         }
         STRINGCOPYN(buf, str, n);
-        return 1;
     }
+    return 1;
```

Comment:
Not deep enough in f2py, but I really doubt that this is the right thing to fix.  This guards against some nonsense object being passed where we cannot pass the result back.

What is even passed in your case even?  Do we have other places where this "optional" stuff works to check what should happen?

Reply:
This function copies the content from the C char buffer to the Python object buffer. In the case where the Python object is `None`, returning without doing anything seems to me to be the right thing to do.

It is `Py_None` that is passed in this case.

This optional thing works fine for other types than strings. In those cases the equivalent function returns early if `PyArray_Check(obj)` is false:  https://github.com/numpy/numpy/blob/54238c7be10104d506259d431e7ce81483666ade/numpy/f2py/cfuncs.py#L306C1-L306C46
Maybe I should make it return -1 instead of 1 for consistency (I'm updating the PR).